src/lib/mysql-helper/lib/connection-manager/index.js

```diff
@@ -108,126 +110,112 @@ class ConnectionManager {
+                return Q.ninvoke(this.connection, 'query', sql, values)
+            })
             .then((queryCallbackArgs) => {
                 return queryCallbackArgs[0]
             })
     }
 
-    rollback(waitTime) {
-        const wait = (ms) => {
-            ms = ms || 500
-            const start = new Date().getTime()
-            for (let i =0; i < 1e7; i++) {
-                if ((new Date().getTime() - start) > ms) {
-                    break;
-                }
-            }
-        }
-
-        wait(waitTime)
-
-        console.log("Rolling back..")
-        return Q.ninvoke(this.connection, 'rollback')
-            .catch((error) => {
-                console.log(error.message)
+    runTransactionQueries(queries) {
+        console.log('mysql-helper:runTransactionQueries')
+        return this.createConnection()
+            .then(() => {
+                return Q.ninvoke(this.connection, 'query', queries.sql.join('; '), queries.values)
             })
-    }
-
-    commit() {
-        return Q.ninvoke(this.connection, 'commit')
-    }
-
-    runTransactionQueries(queryArr, options) {
-        const opts = Object.assign({}, options)
-        opts.groupEvery = options.groupEvery || 1
-
-        const skippedErrMsgs = []
-        const newQueryArr = queryArr
-
-        return queryArr.reduce((promise, query) => {
-            return promise
-                .then(() => {
-                    return this.runQuery(query.sql, [query.values])
-                })
-                .then((response) => {
-                    if (response.serverStatus === 35
-                        && response.affectedRows === 0) {
-                        throw new Error("Not enough phony values in the database.")
+            .then((response) => {
+                const result = response[0]
+                const duplicateIndexes = []
+
+                result.forEach((item, index) => {
+                    if (item.serverStatus === 43 &&
+                        item.affectedRows === 0 &&
+                        item.message === '&Records: 1  Duplicates: 1  Warnings: 0') {
+                        duplicateIndexes.push(index)
                     }
-                })
-                .catch((error) => {
-                    if (error.code !== options.skipErrorCode) {
-                        throw error
+                    else if (item.serverStatus === 43 &&
+                        item.affectedRows === 0 &&
+                        item.message === '&Records: 0  Duplicates: 0  Warnings: 0') {
+                        throw new Error('Insufficient phony values in the database.')
+                    }
+                    else if (item.serverStatus === 3 &&
+                        item.affectedRows === 0 &&
+                        !duplicateIndexes.includes((index-1))) {
+                        throw new Error('Not able to delete phony value. ' +
+                            'May have been used and deleted by another process.')
```

Comment:
This makes me nervous.  Should this even be possible?

Reply:
True.. I added the check here because mysql will not throw an exception if it is not able to delete the row but I cannot think of a case when this would happen.    Will probably remove this check..